spec/registry.md

```diff
@@ -359,6 +359,14 @@ the `:string` selector performs as described below.
 
 The `:string` function returns the string value of the resolved value of the _operand_.
 
+#### Composition
+
+When an _operand_ or an _option_ value uses a _variable_ with a _declaration_
+using an _expression_ with a `:string` _annotation_,
+its resolved value contains only the string value of the _operand_ of the annotated _expression_,
+together with its resolved locale and directionality,
+and no _option_ values.
```

Comment:
I think the wording here is akward because we're trying (rightly so!) to avoid defining any concrete interfaces. However, I note that we do have some requirements about what these interfaces should look like. How about we document these *requirements* instead? Something like:

```suggestion
The return type of the `:string` annotation is implementation-defined.
It MUST meet the following requirements:
- Expose the string value of the operand.
- Expose the locale of the operand.
- Expose the directionality of the operand.
```

Reply:
The proposed text does not sufficiently restrict an implementation when considering the corner cases. For example, in
```
.local $x = {foo :string bar=42}
{{x is {$x :custom}}}
```
it should be clear that `:custom` does not have access to the `bar=42` option value.

Rather than iterating on the exact language within this PR, I'd be much happier if we could find agreement in the proposed _behaviour_ of `:string` values. The main point of this exercise is to define those, rather than lock down absolutely the language describing them.